packages/react-instantsearch/src/core/createConnector.test.js

```diff
@@ -381,6 +383,54 @@ describe('createConnector', () => {
       expect(outputSP).toBe(sp);
     });
 
+    it('calls onSearchParameters when mounted if getSearchParameters are defined', () => {
+      const getSearchParameters = jest.fn(() => null);
+      const onSearchParameters = jest.fn(() => null);
+      let Connected = createConnector({
+        displayName: 'CoolConnector',
+        getProvidedProps: () => null,
+        getSearchParameters,
+        getId,
+      })(() => null);
+      const state = {
+        widgets: {},
+      };
+      const registerWidget = jest.fn();
+      const props = { hello: 'there' };
+      const context = {
+        ais: {
+          store: {
+            getState: () => state,
+            subscribe: () => null,
+          },
+          onSearchParameters,
+          widgetsManager: {
+            registerWidget,
+          },
+        },
+      };
+      mount(<Connected {...props} />, {
+        context,
+      });
+
+      expect(onSearchParameters.mock.calls.length).toBe(1);
+      expect(onSearchParameters.mock.calls[0][0]).toBe(getSearchParameters);
+      expect(onSearchParameters.mock.calls[0][1]).toEqual(context);
+      expect(onSearchParameters.mock.calls[0][2]).toEqual(props);
+
+      Connected = createConnector({
+        displayName: 'CoolConnector',
+        getProvidedProps: () => null,
+        getId,
+      })(() => null);
+
+      mount(<Connected {...props} />, {
+        context,
+      });
+
+      expect(onSearchParameters.mock.calls.length).toBe(1);
```

Comment:
You can use [`.toHaveBeenCalledTimes(1)`](http://facebook.github.io/jest/docs/expect.html#tohavebeencalledtimesnumber)